new feature: tree-like conversation

littlemoonstones opened the pull request:
I have added a new feature: a tree-like conversation, but my code is a little bit messy and there is a bug that causes the loading animation to show immediately. (Fixed it in https://github.com/mckaywrigley/chatbot-ui/commit/fc3f3cf6ef2b92aaee8bdf61bb2d5c8c2b05e675)

https://user-images.githubusercontent.com/32943414/229262252-3533d692-9eb6-4081-994b-636d14ef69ac.mov

nauxliu committed:
Apply the gradient bg style to input on mobile (#77)

nauxliu committed:
Update the Stop Generating button style (#79)

mckaywrigley committed:
fix light mode

ethereumdegen committed:
improve error msg (#81)

Co-authored-by: apexmfer <apexmfer@gmail.com>

mckaywrigley committed:
add dist

CH4R4F committed:
Name conversations automatically #86 (#90)

CH4R4F committed:
fix style issue after clicking on ClearConversation (#89)

there is a small difference in the spaces of the button after confirming which leads to an unexpected animation in the sidebar settings

patanjalikr13 committed:
added download button, downlaod handler and necessary utility functions (#85)

Co-authored-by: Patanjali Kumar <patanjali@oddup.com>

aekeus committed:
Search within conversation.messages + conversation.name (#91)

mckaywrigley committed:
update readme

mckaywrigley committed:
fix scroll behavior

mckaywrigley committed:
tweak scroll tolerance

mckaywrigley committed:
handle code copy error

aweshchoudhary committed:
SEO Fixed and Added Semantic Html Tags (#98)

mckaywrigley committed:
new load behavior

mckaywrigley committed:
fix rogue messages

jean-jacket committed:
autofocus textarea (#103)

alanpog committed:
include prompt in token count (#104)

mckaywrigley committed:
folders (#108)

* folders

* fixes

* storage fix

mckaywrigley committed:
hotfix import

mckaywrigley committed:
stray log

mckaywrigley committed:
clean history on import

mckaywrigley committed:
Update README.md

Chanzhaoyu committed:
feat: show language name (#113)

bcullman committed:
Update Sidebar Setting elements to be buttons (improve accessibility) (#130)

* update sidebar elements to be buttons so they are keyboard navigatable

* rollback whitespace change

logycon committed:
add k8s config (#117)

mckaywrigley committed:
fix import issue

Commit:
feat: codeblock styling like chatgpt (#132)

mckaywrigley committed:
edit message

mckaywrigley committed:
handle mobile edit

ernestobarrera committed:
Added character count to alert message (#135)

This improvement adds additional information to the alert that is displayed when the user enters more characters than allowed in the rich text element. Now, the alert also shows the number of characters that the user has entered, which helps to provide clearer and more useful feedback to the use

aweshchoudhary committed:
Added Better Closing Sidebar Logic (Close on Background Click) (#122)

mckaywrigley committed:
fix sidebar arrow showing

thomasleveil committed:
Do not rely on user to figure out if server is configured with an api key (#136)

Fixes #105 (and probably also #115)

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

mckaywrigley committed:
fix mobile edit

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

Commit:
fix: disable button if messageContent is empty (#137)

mckaywrigley committed:
regenerate button (#138)

mckaywrigley committed:
fix send params

thomasleveil committed:
smoother autoscroll down (#141)

Commit:
fix: change message edit logic (#144)

if the edited message isn't different from the original message, then do nothing.

mckaywrigley committed:
Update README.md

Commit:
feat: Add i18n support for Chinese language (#142)

* feat: Add i18n support for Chinese language

* fix: locale not working in Docker environment

mckaywrigley committed:
hotfix

thomasleveil committed:
Precise error messages (#150)

* Introduce a component to display error messages

* precise error message when api key is invalid

Commit:
feat: Russian language (#148)

* feat: russian (Русский) language

* chore: correct translate

* chore: correct "System prompt" translate

thomasleveil committed:
Locale fr (#153)

* :globe_with_meridians: fr translations

* :coffin: remove unused translation

Anerco committed:
chore: move deps to dev deps (#140)

thomasleveil committed:
:globe_with_meridians: locale de (#154)

Jungley8 committed:
feat: Allow customization of OpenAI host with environment variable (#152)

This commit modifies the OpenAI host configuration to support customization through an environment variable. This change is particularly useful in scenarios where access to the official OpenAI host is restricted or unavailable, allowing users to configure an alternative host for their specific needs.

matias-casal committed:
locale es (#156)

Co-authored-by: Mati <mcasal@bootweb.com.ar>

riande committed:
Add locale sv (Swedish) (#157)

* add locale sv (Swedish)

* add locale sv (Swedish)

thomasleveil committed:
recover from corrupted conversationHistory data (#162)

If an item from conversationHistory is badly corrupted, skip it instead of crashing the app

p-toni committed:
locate pt (#159)

* locate portuguese

* locate portuguese

tulaseerao committed:
Added Telugu language(South Indian) translations (#161)

brianjckim committed:
locale ko (#166)

Co-authored-by: Brian Kim <brian@brianjckim.com>

hyena459 committed:
add_ja_locales (#172)

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

aesadde committed:
fix: update es translations (#167)

Co-authored-by: aesadde <albertosadde@gmail.com>

bcullman committed:
sort the data, not the UI. buttons are now tab-able from the top down (#176)

thejackwu committed:
feat: Message copy button (#171)

* Add copy button

* Fix copy button not copying the entire message

* fix style

* remove prewrap

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

bruce-shi committed:
Latex plugin (#165)

mckaywrigley committed:
hotfix

Commit:
Locale vi (#177)

* Locale vi

* Fix translates layout

hyena459 committed:
Corrected wording (#178)

srsholmes committed:
feat: add in prettier and format code for consistency (#168)

Chanzhaoyu committed:
feat: export json format (#185)

HaithamLeo committed:
added CONTRIBUTING.md file (#188)

Commit:
chore: delete code light theme, like chatgpt (#186)

Commit:
chore: change sidebar font size & style (#191)

* chore: change sidebar font size & style

* chore: create font size style for sidebar

oznav2 committed:
add_Hebrew_locales (#190)

* add_Hebrew_locales

Add Locales of Hebrew

* Update next-i18next.config.js

updating i-18next with Hebrew

Commit:
fix: fix fonts (#194)

* fix: hotfix fonts

* chore: set normal line height

nahid18 committed:
feat: bangla lang support (#195)

Co-authored-by: nahid18 <nahidpatwary50@email.com>

umarhadi committed:
feat: add Bahasa Indonesia support (#198)

Signed-off-by: umarhadi <hi@umarhadi.dev>

tatsuiman committed:
docker build action (#196)

jiayao committed:
Allow switching GPT model in the middle of a conversation (#181)

* Allow switching model in the middle of a conversation

* Hide model selection menu behind a settings button

* rename

* Touch up the settings icon

Commit:
chore: restyle modal with model select (#202)

chore: set normal font size for sidebar
chore: set normal gradient for `ChatInput`

Commit:
Fix unreadable dropdown menu text in dark mode (#204)

* Fix unreadable dropdown menu text in dark mode

* Fix unreadable dropdown menu text in dark mode

Commit:
fix: missing translation (#206)

for translators: need translate `Save & Submit` and `Cancel`

DannyAziz committed:
Add some additional text for API Key (#210)

* feat: API Key Link

* feat: key input placeholder

bcullman committed:
convert conversation icons to buttons (accessibility) (#192)

* update sidebar overflow

* update all clickable icons to buttons

* refactor buttons so they are not inside other buttons

* format doc

* update input background to transparent

* adjust btns size to match #202

* update text size per #202

thomasleveil committed:
Fix rendering performances issues related to scrolling events (#174)

* memoize chat related components

* Avoid re-rendering ChatInput on every message udpate

* change the way the horizontal scrollbar is hidden

* make the scroll event listener passive

* perf(Chat): fix performances issues related to autoscroll

Uses the intersection API to determine autoscroll mode instead of listening for scroll events

* tuning detection of autoscroll

chinleez committed:
Display optimization on iPhone device page (#220)

* Disable Zoom Page

* add height

---------

Co-authored-by: Praise <lizan60@gmail.com>

mckaywrigley committed:
Option for clear all messages (#222)

Co-authored-by: ryanhex53 <ouyang.em@gmail.com>

pujux committed:
fix german wording and switch to informal style (#226)

Commit:
chore: some small improvements (#223)

* chore: stylize error message div
chore: correct styles for sidebar btn
chore: add spinner and replace header "Please wait" on spinner
chore: correct Russian translate
chore: hide clear conversation btn if not conversations
chore: stylize "Need OpenAI key" div

* chore: corrent Russian translate

mckaywrigley committed:
fix regenerate

Commit:
chore: fix styles in "OpenAI Key" div (#228)

chore: do nothing if the name of the renamed dialog contains nothing.
chore: fix some styles
chore: remake conversation settings on mobile

mckaywrigley committed:
Prompts (#229)

SyedMuzamilM committed:
fix: overlap between plus and prompt menu icon (#230)

thomasleveil committed:
fix import  (#242)

* :bug: fix import (#224)

* :bug: fix import of corrupted history

see https://github.com/mckaywrigley/chatbot-ui/issues/224#issuecomment-1486080888

* add the run-test-suite github action

bcullman committed:
convert folders to buttons & folder icons to buttons (accessibility)  (#237)

* tabbable folders

* fix spacing

bcullman committed:
update prompts to be tabbable (#241)

Chanzhaoyu committed:
feat: empty state add icons (#248)

SyedMuzamilM committed:
fix: prompt sidebar not showing on small devices (#232)

* fix: overlap between plus and prompt menu icon

* fix: prompt sidebar not showing on small devices

bcullman committed:
make all chat area components tabbable (accessibility) (#246)

* make all chat area components tabbable

* align message role description

* remove inline styles on icons

* remove inline styles on icons

Chanzhaoyu committed:
chore: update chinese locales (#247)

* chore: update chinese locales

* chore: update locales

liby committed:
fix: resolve Enter event conflict during CJK IME (#253)

* fix: resolve Enter event conflict during CJK IME

* add

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

mckaywrigley committed:
scroll btn (#256)

* feat: added scroll down button

when the use scrolls up on the chat a button will appear on the bottom right side of the screen. It will smoothly scroll down to the bottom of the chat when the button was pressed.

* remove env

---------

Co-authored-by: dasunNimantha <dasun4@pm.me>

mckaywrigley committed:
Update README.md

srsholmes committed:
chore: migrate to vitest for 10x faster tests (#257)

* chore migrate to vitest

* chore: cleanup jest stuff

* chore: install the coverage dep

SyedMuzamilM committed:
fix: layout shifting on small devices (#258)

* fix: overlap between plus and prompt menu icon

* fix: prompt sidebar not showing on small devices

* fix: layout shifting on small devices when there is a code block

dasunNimantha committed:
style: changed scroll icon size via size attribute as suggested (#263)

SyedMuzamilM committed:
fix: website crash when typing / followed with non prompt name (#262)

* fix: froward slash crash

* added the rounded corners when editing folder/chat

Commit:
doc: update doc about Docker (#264)

maybe fix #259

ernestobarrera committed:
New JSON translation file for Spanish (#270)

Translation to Spanish of the prompts custom template bar

DjTer3a committed:
Adding arabic translation (#274)

thecrazybob committed:
Fix invalid Tailwind property (#275)

dasunNimantha committed:
chore: added Sinhala language locales (#276)

thomasleveil committed:
:sparkles: feat: add DEFAULT_MODEL environment variable (#280)

* :sparkles: feat: add DEFAULT_MODEL environment variable

* set the model maxLength setting in the models definition

* set the model tokenLimit setting in the models definition

Chanzhaoyu committed:
fix: translation omission (#281)

mckaywrigley committed:
Update README.md

mckaywrigley committed:
change messaging

dasunNimantha committed:
fix: In light mode, the arrow in the "scroll to bottom" icon is difficult to see #299 (#304)

ryanhex53 committed:
return detail error from api call. (#295)

burke committed:
Allow specifying OpenAI-Organization header: (#313)

See https://platform.openai.com/docs/api-reference/introduction

> For users who belong to multiple organizations, you can pass a header
> to specify which organization is used for an API request. Usage from
> these API requests will count against the specified organization's
> subscription quota.

SyedMuzamilM committed:
fix: small ui issue (#311)

* fix: unnecessary lines before the chats or prompts

Removed unncessary lines if there are no folders before prompts and
chats.

* added the light mode for scroll down button

MatriQ committed:
fix: properly trigger prompt modal close (#309)

Co-authored-by: MatriQi <matri@aifi.io>

liby committed:
feat: support custom default system prompt (#285)

mckaywrigley committed:
change default prompt in example

mckaywrigley committed:
fix nasty bug

mckaywrigley committed:
update demo

rudolfolah committed:
Feature: add link to openai account usage (#319)

* Update ModelSelect.tsx to include link to account usage

It would be helpful to be able to click a link to see the account usage and billing, and the model selection is related to the pricing.

* updated styling and added icon for external link

* fixed missing import

brijeshb42 committed:
Autofocus input when after clicking OpenAI API Key in sidebar. (#324)

* Update Key.tsx

Autofocus input after its visible.

* Update Key.tsx

colindotfun committed:
fix: label layout shift (#338)

littlemoonstones committed:
new feature: tree-like conversation

mckaywrigley commented:
This is *really* cool. We'll definitely merge this idea in.

It needs just a bit of work around UI/UX and making sure it doesn't break any core things.

I'll bump this up on my todo list. Great start here.

littlemoonstones commented:
Sorry, I have found the reason that I accidentally deleted some lines of code. Here's the link to the specific lines that were deleted: https://github.com/mckaywrigley/chatbot-ui/blob/aefec17525b664ba3e2d82512fa0196622925182/pages/index.tsx#L165-L199

Commit:
fix: prompts list height (#340)

danielkhoo committed:
Fix icon color (#344)

bcullman commented on pages/index.tsx:
```diff
@@ -528,12 +500,12 @@ const Home: React.FC<HomeProps> = ({
-      setCurrentMessage(undefined);
-    }
-  }, [currentMessage]);
+  // useEffect(() => {
```
can this be removed?

littlemoonstones replied:
Hi, I've addressed your comment and left the commented useEffect in the code. Please check the changes in commit c676f6fcf09b02831472d06e39d1c07a1d8f427d.

bcullman commented on utils/contexts/conversaionContext.tsx:
this filename is misspelled.

littlemoonstones replied:
Thanks, I rename it in commit https://github.com/mckaywrigley/chatbot-ui/commit/fc9b66b9aa8d78483f30c6f7f037aa218d92c44f

mckaywrigley committed:
fix spacing

littlemoonstones committed:
Remove commented useEffect for handling currentMessage

littlemoonstones committed:
correct the file name

riande committed:
locale improvements (#351)

aaliboyev committed:
fix for openai token limit error (#350)

superman66 committed:
feat: support import and export with prompts (#330)

* feat: support import and export prompts

* test: update importExports.test.ts

* Delete .gitpod.yml

mckaywrigley committed:
change to "data"

Jdban committed:
Adds a loading spinner that replaces send icon while result is generated (#329)

Jdban committed:
add react-hot-toast and surface OpenAI API errors to users (#328)

SyedMuzamilM committed:
fix: some ui issues (#346)

* fix: scroll button not visible in light mode

* fix: sidebar when there is a folder

When a folder is added in the sidebar and there are less items scroll
bar appears. This simple change fixes that behaviour

* fix: small devices regerate/Stop button below input

Below 768px Stop Genrating and Regerate Button remains hidden behind the
input. This is the fix for that

bcullman committed:
remove dupe main (#355)

bcullman committed:
remove unused code (#356)

mckaywrigley committed:
change output limit

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

spctechdev committed:
Locale it (#363)

* Add files via upload

* Update next-i18next.config.js

* Update chat.json

mckaywrigley committed:
revert

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

alanpog committed:
tokenLimit now can be closer to real model limit. (#366)

thomasleveil committed:
Fix translations import export (#376)

* fix "Import data" case

* fix "Export data" translations

thomasleveil committed:
:notes document (#373)

jpzhangvincent committed:
update chat role icon (#378)

bcullman commented:
@littlemoonstones does this effect import/export data?

Commit:
Update Vietnamese (#385)

spctechdev committed:
Fix previous IT locale (#374)

* Update chat.json

* Update promptbar.json

* Update sidebar.json

* Update sidebar.json

* Update sidebar.json

littlemoonstones commented:
> @littlemoonstones does this effect import/export data?

I have tested it and found that it does not affect the import/export of data.

oznav2 committed:
Update promptbar.json (#389)

fix : added missing translation

RossHartmann commented:
> @littlemoonstones does this effect import/export data?

@bcullman I haven't yet tested on this branch but I suspect that after we make this modeling change, it would affect importing of any previous data formats since `Conversation` will no longer have a `messages` property -- while previous data formats *did* have a `messages` property.

I'll create a new data format to handle import/export of this new format in #377 then we can merge that into this branch.

UPDATE: I'm going to need to do the model changes on this branch since the changes greatly affect UI elements.

RossHartmann committed:
Merge branch 'upstream/main' into littlemoonstones/feature/update-conversation-model-RH

# Conflicts:
#	components/Chat/ChatMessage.tsx
#	pages/_app.tsx
#	pages/index.tsx

RossHartmann committed:
- Add new Export Format for new Conversation model where messages were replaced with ChatNodes
- Modify the cleaning process for data formats. Instead of trying to convert each format to the latest format, we now iteratively upgrade each format (from V1 to V2, then V2 to V3, etc.) This makes it so that when we add a new format, we only need to upgrade the last format to the new format (rather than all previous formats to the new format).

RossHartmann committed:
- Add "cleaning v3 data" test
- Add "cleaning v4 data" test.

RossHartmann committed:
Cleaning history / Conversation for all formats

RossHartmann committed:
Add new Message format to more closely align with ChatGPT's model

RossHartmann commented:
@littlemoonstones I've created a [PR in your fork](https://github.com/littlemoonstones/chatbot-ui/pull/1) to merge into this branch. It updates the Conversation model and brings in the latest changes from `main`. Please let me know if you want to review anything.

littlemoonstones committed:
Merge pull request #1 from RossHartmann/littlemoonstones/feature/update-conversation-model-RH

Add V5 Conversation Model for Tree-Like Conversations

RossHartmann commented:
@bcullman Would you mind giving me more detail around why the Vercel bot failed to deploy to Preview? Looks like all unit tests are passing, but I can't see any details around the deploy failure.

RossHartmann committed:
Fix send message

RossHartmann commented:
Thank you, @littlemoonstones! Just created [another PR](https://github.com/littlemoonstones/chatbot-ui/pull/2) to fix this -- build and test both now pass.

littlemoonstones committed:
Merge pull request #2 from RossHartmann/littlemoonstones/bugfix/fix-send-message-RH

Fix Build Failure

RossHartmann committed:
Fix cleaning of selected conversation

RossHartmann committed:
Check to ensure prompt, model, and folderId have a value

littlemoonstones committed:
Merge pull request #3 from RossHartmann/littlemoonstones/bugfix/fix-clean-selected-conversation-RH

Fix Cleaning of Selected Conversation

RossHartmann committed:
Fix issue where first message was a non-system message and wasn't being displayed.

littlemoonstones committed:
Merge pull request #4 from RossHartmann/littlemoonstones/bugfix/fix-missing-starting-message-RH

Display Non-System First Message

kaaboaye commented:
I think it might be better not to show those `<1/1>` and only show them after prompt was modified and there is something to do with those buttons.

If `<1/1>` were to be shown to all users every time it would create a lot of confusion especially that those buttons do not provide any value unless prompt was modified and there is `<2/2>`.

<img width="147" alt="image" src="https://user-images.githubusercontent.com/12039839/229745546-9cfa47cf-92ad-4ae2-9c68-f4124cc8969b.png">

bhaskoro-muthohar commented:
When can this feature be merged, guys? I need it so bad 🚀

kaaboaye commented:
> When can this feature be merged, guys? I need it so bad 🚀

You can always run this brunch with tree-like directly instead waiting for merge.

littlemoonstones committed:
Fixed message streaming, but there is still a remaining bug: cannot store into the conversation

littlemoonstones commented:
> Sorry, I have found the reason that I accidentally deleted some lines of code. Here's the link to the specific lines that were deleted:
> 
> https://github.com/mckaywrigley/chatbot-ui/blob/aefec17525b664ba3e2d82512fa0196622925182/pages/index.tsx#L165-L199

I am trying to fix the message streaming in commit d8af3c7. Currently, it can display the streaming message successfully, but after the entire message shows up, the `selectedConversation` cannot be saved into localStorage. 



https://user-images.githubusercontent.com/32943414/229780827-9b9f8e9a-2906-4c4f-9d0d-4f0668d9899f.mov




The process is roughly like the diagram below:

```mermaid
sequenceDiagram
    participant handleSend
    participant selectedConversation
    participant chatNode_A
    participant messageStreaming
    participant saveConversation

    handleSend->>selectedConversation: call handleSend
    selectedConversation->>chatNode_A: add new chatNode(A)
    loop message streaming process
        messageStreaming->>chatNode_A: update content with text
    end
    chatNode_A->>saveConversation: call saveConversation
```
I guess the reason is related to asynchrony, but I cannot resolve the problem. Could anyone help me?

littlemoonstones commented:
> I think it might be better not to show those `<1/1>` and only show them after prompt was modified and there is something to do with those buttons.
> 
> If `<1/1>` were to be shown to all users every time it would create a lot of confusion especially that those buttons do not provide any value unless prompt was modified and there is `<2/2>`.
> 
> <img alt="image" width="147" src="https://user-images.githubusercontent.com/12039839/229745546-9cfa47cf-92ad-4ae2-9c68-f4124cc8969b.png">

OK, I have adjusted it in [cda77b5](https://github.com/mckaywrigley/chatbot-ui/pull/339/commits/cda77b58117a2d04924582c0cd0c101fe9c469f4).
<img width="83" alt="image" src="https://user-images.githubusercontent.com/32943414/229825034-5712e890-3382-45b8-86d4-c7db721cdcdf.png">

littlemoonstones committed:
fixed the bug: streaming message

littlemoonstones committed:
remove "console.log() for debug"

littlemoonstones commented:
> > Sorry, I have found the reason that I accidentally deleted some lines of code. Here's the link to the specific lines that were deleted:
> > https://github.com/mckaywrigley/chatbot-ui/blob/aefec17525b664ba3e2d82512fa0196622925182/pages/index.tsx#L165-L199
> 
> I am trying to fix the message streaming in commit [d8af3c7](https://github.com/mckaywrigley/chatbot-ui/commit/d8af3c7226c31f37d045e23c7a9a21e20bf7d247). Currently, it can display the streaming message successfully, but after the entire message shows up, the `selectedConversation` cannot be saved into localStorage.
> 
> The process is roughly like the diagram below:
> 
> 
> I guess the reason is related to asynchrony, but I cannot resolve the problem. Could anyone help me?

Fixed in fc3f3cf

littlemoonstones committed:
fixed: missing some variables when sending chatNode

littlemoonstones committed:
Merge branch 'develop'

littlemoonstones committed:
fix the type of sending messages

littlemoonstones committed:
hide switch button if the number of messages less than 1

mckaywrigley committed:
push (#414)

mckaywrigley committed:
Google fixes (#415)

* push

* fixes

mckaywrigley committed:
brevity

mckaywrigley committed:
spelling

RossHartmann committed:
Merge branch 'upstream/main' into littlemoonstones/feature/merge-in-main-RH

# Conflicts:
#	components/Chat/Chat.tsx
#	components/Chat/ChatInput.tsx
#	pages/index.tsx

RossHartmann committed:
Fixed issue setting current node

itbm committed:
Hide API Key and Plugin Key buttons if set server side (#423)

thomasleveil committed:
update README regarding GOOGLE_API_KEY and GOOGLE_CSE_ID (#422)

thomasleveil committed:
Fix 416 copy button on code block (#427)

* update README regarding GOOGLE_API_KEY and GOOGLE_CSE_ID

* :bug: fix copy button not showing up on code block with no recognized language

fixes #416

littlemoonstones committed:
Merge pull request #5 from RossHartmann/littlemoonstones/feature/merge-in-main-RH

Merge in Main's Plugin Changes

RossHartmann committed:
Merge branch 'upstream/main' into littlemoonstones/fix-v5-import-RH

# Conflicts:
#	components/Chat/ChatMessage.tsx
#	pages/api/google.ts

RossHartmann committed:
Fix importing v5 data

littlemoonstones committed:
Merge pull request #6 from RossHartmann/littlemoonstones/fix-v5-import-RH

RossHartmann committed:
Fix Google search

RossHartmann committed:
Remove unncessary lines

littlemoonstones committed:
Merge pull request #7 from RossHartmann/littlemoonstones/bugfix/fix-google-search-RH

RossHartmann commented:
Anything we have left to do on this PR? @littlemoonstones

littlemoonstones commented:
> Anything we have left to do on this PR? @littlemoonstones

The UI needs optimization, but I'm not good at it. I wonder if it's necessary to optimize the UI for this PR?

RossHartmann committed:
Update positioning of thread/tree indicator

littlemoonstones committed:
Merge pull request #8 from RossHartmann/littlemoonstones/feature/display-update-RH

Update Positioning Tree Indicator

RossHartmann commented:
@mckaywrigley We've done a few UI optimizations. Any other changes you'd like to see on this PR before merging?

RossHartmann committed:
Mimic UI of ChatGPT

littlemoonstones committed:
adjust UI

littlemoonstones committed:
Merge pull request #9 from RossHartmann/littlemoonstones/feature/mimic-chatgpt-tree-RH

Mimic Tree UI of ChatGPT

littlemoonstones committed:
fixed the ChatSwitch.displayName

mckaywrigley committed:
MAJOR REFACTOR (#494)

* move index to home folder, create state and context files and barrell folder

* Sanity Check Commit:  reducer added to home.tsx manual QA all working

* WIP: promptBar

* fix missing json parse on folders and prompts

* split context and add promptbar context

* add context to nested prompt componets and componetize Folder componet

* remove log

* Create buttons folder and componetize sidebar action button

* tidy up prompt handlers

* componetized sidebar

* added back chatbar componet to left side sidebar

* monster commit: Componetized the common code between chatbar and promptbar into new componet Sidebar and added context to both bars

* add useFetch service

* added prettier import sort to keep imports ordered and easier to indentify

* added react query and useFetch to work with RQ

* added apiService, errorService and reactQuery

* add callback and tidy up error service

* refactor chat and child componets to useContext

* fix extra calls and bad calls to mel endpoint

* minor import cleanup

---------

Co-authored-by: jc.durbin <jc.durbin@ardanis.com>

akashkriplani committed:
fix parameter of length to type "number" (#492)

dotneet committed:
fix google plugin bug due to URI encoding issue. (#469)

* fix: uri encoding

* fix: error handling

MKTHINGS committed:
Add Romanian translation (#468)

* Create Romanian chat.json

I have created Romanian translation of the chat.json.

* Add Romanian translation

I have created Romanian translations.

* Update next-i18next.config.js

Added ro locale

---------

Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

regisrex committed:
Update demo image (#459)

thomasleveil committed:
document OPENAI_API_HOST and OPENAI_ORGANIZATION (#429)

dornfeld-capital committed:
improve plugin hotkey support (#426)

itbm committed:
Add support for Azure OpenAI (#495)

mckaywrigley committed:
log removal

jony4 committed:
fix: issues/498 (#499)

Klaudioz committed:
Update README.md (#521)

Fix no existing route to index.ts

fox91 committed:
Added support to docker multi-platform image, close #518 (#519)

SyedMuzamilM committed:
fix: modal scrolling on small devices (#527)

* fix: propmt side not showing on small devices

* added the robot icon while chat is loading

* fix: sidbar position

* fix: modal not scrolling on small devices

* fix: overlap between the modal and icons

mckaywrigley committed:
delete message (#531)

* feat: delete single message

* improve plugin hotkey support (#426)

* feat: delete single message

* remove logs

* remove logs

---------

Co-authored-by: Ryland <rylandl@qq.com>
Co-authored-by: Dornfeld Capital <75278543+dornfeld-capital@users.noreply.github.com>
Co-authored-by: L.Ryland <41134883+L-Ryland@users.noreply.github.com>

ivanfioravanti committed:
Feature request: Adding temperature as parameter (#513)

* Adding temperature as parameter

* NEXT_PUBLIC_ prefix added

* add spacing

---------

Co-authored-by: Ivan Fioravanti <>
Co-authored-by: Mckay Wrigley <mckaywrigley@gmail.com>

adiestel committed:
Fix for DEFAULT_SYSTEM_PROMPT (#510)

chrisociepa committed:
Add Polish translation (#532)

* Add Polish translation

* Minor fix in Polish translation

clevermiraz committed:
Update Bangla Translation (#546)

adiestel committed:
Fix: Promptbar initializes in open state for Mobile (#542)

* Fix for DEFAULT_SYSTEM_PROMPT

* Set the prompt bar to be hidden if screen < 640

mckaywrigley committed:
update readme

MalTarDesigns committed:
Fix search box disappearing in Sidebar component (#592)

Co-authored-by: Jamal Johnson <you@example.com>

liby committed:
Fix message input not updating on chat list switch (#590)

borborborja committed:
New language, catalan (#587)

KhanKudo committed:
Fixed: site cannot open if imported json invalid (#579)

mirkan1 committed:
Turkish language implementation for i18n (#576)

* turkish language support implemented

* tr as string in module declaration

* missing key Plugin Keys implementation

* synonyms is used to get rid of confusion and to avoid repetition. The word “synonym” is derived from the Greek word “syn” meaning “together” and “onym” meaning “name” while in Turkish name and noun can be translated as both "ad" and "isim"

* //

shemarlindie committed:
fix: chat messages disappear on edit (#575)

Resolves #574

dotneet committed:
feat: settings dialog and moved theme settings to dialog from sidebar (#570)

* feat: settings dialog and moved theme settings to dialog from sidebar.

* chore(locale): move some labels to settings from sidebar

dotneet committed:
chore: use last conversation temperature (#568)

dotneet committed:
fix: temperature font color (#556)

santi4o committed:
fix: sroll down button overlaping chat input (#564)

mckaywrigley committed:
readme update

mckaywrigley committed:
Merge branch 'main' of https://github.com/mckaywrigley/chatbot-ui-pro

dotneet committed:
chore: improve Japanese language locales (#554)

mckaywrigley committed:
Add temp display (#596)

* Made the Model Name Bar Sticky at the top of the Chat Window

* Added Temp to the top bar display

* barliens

---------

Co-authored-by: Chris Wall <cwall@wooldata.com>

mckaywrigley committed:
Made the Model Name Bar Sticky at the top of the Chat Window (#597)

Co-authored-by: Chris Wall <cwall@wooldata.com>

borborborja committed:
add catalan language (#598)

mckaywrigley committed:
import/export add instead of replace (#601)

* change search language

* add to import/export

* fix log

mckaywrigley committed:
sort folders alphabetically

mckaywrigley committed:
change empty msg

ryanhex53 committed:
more chinese translation. (#609)

shemarlindie committed:
fix: optimize mobile layout (#606)

- conserve space in new convo view
- fix overlapping edit and delete chat message icons
- fix overlapping copy icon

bigsk1 committed:
Create Security.md (#607)

basic secuirty policy as the repo didn't have one yet, update as needed.

anthonypuppo committed:
Performance (#602)

mckaywrigley committed:
readme update

mckaywrigley committed:
fix prompt selection

mckaywrigley committed:
FIX IMPORT

mckaywrigley committed:
force update

kyyhkynen committed:
add Finnish translations (#618)

anthonypuppo committed:
Add blinking typing cursing (#619)

moe001 commented:
Is there any new progress on this job?

RossHartmann commented:
> Is there any new progress on this job?

@moe001 I suspect this PR needs to go through some fairly massive changes due to the recent refactor in #494.

littlemoonstones commented:
I have started to modify the code.

littlemoonstones committed:
work, but import/export need fixed

littlemoonstones committed:
fixed import/export

littlemoonstones committed:
format the code

littlemoonstones committed:
fiexed clear all messages

littlemoonstones committed:
remove the code related debug

littlemoonstones committed:
Merge remote-tracking branch 'upstream/main' into refactor_merging

littlemoonstones committed:
remove redundant file

littlemoonstones committed:
Merge remote-tracking branch 'upstream/main'

littlemoonstones commented:
Currently, I commented on the function `handleDeleteMessage` in 7d990ab because I have no idea how to deal with deleting messages in a tree-like conversation.
